omas/tests/machine_mappings/test_d3d.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+import omas
+import os
+import numpy as np
+from omas.machine_mappings.d3d import ec_launcher_active_hardware
+from omas.omas_machine import tunnel_mds
+
+class TestD3D(unittest.TestCase):
+
+    def test_d3d_ECH_machine_mapping(self):
+        '''
+        Test the machine mapping of the DIII-D ECH against the a .json file created with fetchGyrotron from the OMFIT TORAY module
+        '''
+        pristine_ods = omas.load_omas_json(os.path.join('tests', 'test_data', 'd3d_ECRH_machine_mapping.json'))
+        test_ods = omas.ODS()
+        tunnel_mds('d3d', 'RF')
+        ec_launcher_active_hardware(test_ods, 170325)
+        # test_ods.open('d3d', 170325)
+        fields = ['launching_position.r', 'launching_position.z', 'launching_position.phi', 
+                  'frequency.data', 'power_launched.data', 'mode.data', 'steering_angle_tor.data',
+                  'steering_angle_pol.data']
+        prefix = 'ec_launchers.launcher'
+        for launch in pristine_ods[prefix]:
```

Comment:
You can compare two ods with:
```python
ods1.diff(ods2)
```

Reply:
I don't think this is going to work here. The TORAY generated ods only has one timepoint while the machine mapping routine loads all timepoints. Also, they don't hold the same fields. Some time fields are missing in the TORAY sample file, the time units are in milliseconds and there are some code parameters in the TORAY file that are not loaded with the machine mapping. For the sample, it would be most likely better to only have one timepoint to save git space.

Reply:
@AreWeDreaming time must always be in seconds in an ODS

Reply:
Yes, but this is a bug in the TORAY module. I didn't want to touch that at all so I could use its output for my test. Arguably now that everything has been validated once, it might be better to create an ODS with the new machine mapping routines and use that for future verification. Should I do that?